catroid/src/main/java/org/catrobat/catroid/utils/Utils.java

```diff
@@ -1104,4 +1104,14 @@ public static boolean isUserLoggedIn(Context context) {
 	public static String getNumberStringForBricks(float value) {
 		return (int) value == value ? "" + (int) value : "" + value;
 	}
+
+	public static <T extends Object> List<T> distinctListByClassOfObjects(List<T> listToDistinct) {
+		List<T> distinctList = new ArrayList<>();
+		HashMap<Class, T> uniqueMap = new HashMap<>();
+		for (T objectInstance : listToDistinct) {
+			uniqueMap.put(objectInstance.getClass(), objectInstance);
+		}
+		distinctList.addAll(uniqueMap.values());
+		return distinctList;
```

Comment:
`extends Object` is necessary (IMHO) because i call `Object.getClass()`. Would not be able to do that (accourding to [this question](http://stackoverflow.com/questions/3437897/how-to-get-a-class-instance-of-generics-type-t).
Map as declaration for what exactly?